src/software/ai/navigator/path_planner/theta_star_path_planner.h

```diff
@@ -243,50 +253,66 @@ class ThetaStarPathPlanner : public PathPlanner
      *
      * @return cell in grid
      */
-    Coordinate pointToCoordinate(Point p);
+    Coordinate convertPointToCoord(const Point &p) const;
 
     /**
-     * Try to find a path to destination and leave
+     * Try to find a path to end and leave
      * trail markers along the way
      *
-     * @param dest_coord destination coordinates
+     * @param end_coord end coordinates
      *
-     * @return if path to destination was found
+     * @return true if path to end was found
      */
-    bool findPathToDestination(Coordinate dest_coord);
+    bool findPathToEnd(const Coordinate &end_coord);
 
     /**
-     * Check for invalid or blocked src_coord and dest_coord and
-     * adjust parameters accordingly
+     * Update vertex for all neighbours (all 8 directions) of current_coord
+     *
+     * @param current_coord The current coordinate
+     * @param end_coord end coordinates
+     *
+     * @return true if path to end was found among successors
+     */
+    bool visitSuccessors(const Coordinate &current_coord, const Coordinate &end_coord);
+
+    /**
+     * Adjusts start and end points for navigability and determines if a path cannot be
+     * found
+     *
+     * If start or end are non-navigable, then no path exists
      *
-     * @param src_coord source coordinate
-     * @param dest_coord destination coordinate
+     * If start or end are blocked by obstacle(s), then try to find a nearby point that
+     * isn't blocked
+     * - If a point can be found then set the endpoint to that nearest point
+     * - If no nearby points are unblocked, then no path exists
      *
-     * @return true if there is no path to destination
+     * @param [in/out] start_coord source coordinate
+     * @param [in/out] end_coord end coordinate
+     *
+     * @return true if there is no path to end
      */
-    bool checkForInvalidOrBlockedCases(Coordinate &src_coord, Coordinate &dest_coord);
+    bool adjustEndPointsAndCheckForNoPath(Coordinate &start_coord, Coordinate &end_coord);
 
     /**
-     * Check if start to destination is at least a grid cell away
+     * Check if start to end is within a grid cell or within a constant threshold
```

Comment:
nit: I think would be more clear if the threshold was a parameter, and the `CLOSE_TO_END_THRESHOLD` constant or whichever value is used it passed that way.

Reply:
this function isn't super necessary so I just moved the check back into findPath